security/abac/role/src/main/java/io/helidon/security/abac/role/RoleValidator.java

```diff
@@ -116,6 +116,17 @@ public RoleConfig fromAnnotations(EndpointConfig endpointConfig) {
                         }
                         builder.permitAll(false);
                         builder.denyAll(false);
+                    } else if (annotation instanceof RolesContainer) {
+                        RolesContainer container = (RolesContainer) annotation;
+                        for (Roles role : container.value()) {
+                            if (role.subjectType() == SubjectType.USER) {
+                                roles.addAll(Arrays.asList(role.value()));
+                            } else {
+                                serviceRoles.addAll(Arrays.asList(role.value()));
+                            }
+                        }
+                        builder.permitAll(false);
```

Comment:
we should do this only if the roles container has at least one value (I think it can be declared as empty).

Reply:
done